common/src/main/java/tachyon/ClientBase.java

```diff
@@ -188,4 +191,90 @@ protected synchronized InetSocketAddress getAddress() {
     return mAddress;
   }
 
+  /**
+   * The RPC to be executed in {@link #retryRPC(RpcCallable)}.
+   *
+   * @param <V> the return value of {@link #call()}
+   */
+  protected interface RpcCallable<V> {
+    /**
+     * The task where RPC happens.
+     *
+     * @return RPC result
+     * @throws TException when any exception defined in thrift happens
+     */
+    V call() throws TException;
+  }
+
+  /**
+   * Same with {@link RpcCallable} except that this RPC call throws {@link TachyonTException} and
+   * is to be executed in {@link #retryRPC(RpcCallableThrowsTachyonTException)}.
+   *
+   * @param <V> the return value of {@link #call()}
+   */
+  protected interface RpcCallableThrowsTachyonTException<V> {
+    /**
+     * The task where RPC happens.
+     *
+     * @return RPC result
+     * @throws TachyonTException when any {@link TachyonException} happens during RPC and is wrapped
+     *                           into {@link TachyonTException}
+     * @throws TException when any exception defined in thrift happens
+     */
+    V call() throws TachyonTException, TException;
```

Comment:
But I still think the solution in this PR is fine since the RPCs that won't throw TachyonTException and the ones that does thrown can be considered different RPC types.